packages/webdriverio/src/commands/element/click.js

```diff
@@ -37,11 +37,40 @@
  * </example>
  *
  * @alias element.click
- * @uses protocol/element, protocol/elementIdClick
+ * @uses protocol/element, protocol/elementIdClick, protocol/performActions, protocol/positionClick
  * @type action
- *
+ * @param {Object=} options object containing a property called `button` which can be set to 0 (left) 1 (middle) or 2 (right) button
  */
 
-export default function click () {
-    return this.elementClick(this.elementId)
+export default async function click (options) {
+    const { button } = options || {}
+    const isLeftButton = button === 'left' || button === 0 || !button
+    const isMiddleButton = button === 'middle' || button === 1
+    const isRightButton = button === 'right' || button === 2
+
+    if (isLeftButton) {
+        return this.elementClick(this.elementId)
+    }
+
+    if (isRightButton || isMiddleButton) {
+        if (this.isW3C) {
+            await this.performActions([{
+                type: 'pointer',
+                id: 'pointer1',
+                parameters: { pointerType: 'mouse' },
+                actions: [
+                    { type: 'pointerMove', origin: this, x: 0, y: 0 },
+                    { type: 'pointerDown', button: isRightButton ? 2 : 1 },
```

Comment:
Should we have a bit pause in here? I've seen action commands that use pause to simulate the click interaction

Reply:
This is only used in a doubleClick in between the two clicks so I don't think it is needed. I tested this in Chrome 77 and all buttons function as expected.